src/main/java/com/lambdaworks/redis/RedisURI.java

```diff
@@ -170,10 +172,67 @@ public static RedisURI create(URI uri) {
             }
         }
 
+        if (isNotEmpty(uri.getQuery())) {
+            StringTokenizer st = new StringTokenizer(uri.getQuery(), "&;");
+            while (st.hasMoreTokens()) {
+                String queryParam = st.nextToken().toLowerCase();
+                if (queryParam.startsWith(TIMEOUT_PARAMETER_NAME)) {
+                    parseTimeout(builder, queryParam);
+                }
+            }
+        }
+
         return builder.build();
 
     }
 
+    private static void parseTimeout(Builder builder, String queryParam) {
+        int index = queryParam.indexOf('=');
+        if (index < 0) {
+            return;
+        }
+
+        String timeoutString = queryParam.substring(index + 1);
+
+        int numbersEnd = 0;
+        while (numbersEnd < timeoutString.length() && Character.isDigit(timeoutString.charAt(numbersEnd))) {
+            numbersEnd++;
+        }
+
+        if (numbersEnd == 0) {
+            if (timeoutString.startsWith("-")) {
+                builder.withTimeout(0, TimeUnit.MILLISECONDS);
+            } else {
+                // no-op, leave defaults
+            }
+        } else {
+            long timeoutValue = Long.parseLong(timeoutString, 0, numbersEnd, 10);
```

Comment:
This line causes compile errors, see https://travis-ci.org/mp911de/lettuce/builds/90676460